src/evox/problems/numerical/lsmop.py

```diff
@@ -157,29 +103,31 @@ def evaluate(self, state, X):
         g = jnp.zeros([n, m])
         for i in range(0, m, 2):
             for j in range(1, self.nk + 1):
-                g = g.at[:, i : i + 1].set(
-                    g[:, i : i + 1]
-                    + LSMOP._Sphere(
+                g = g.at[:, i].set(
+                    g[:, i]
+                    + self.sphere.evaluate(
+                        state,
                         X[
                             :,
                             int(self.len_[i] + m - 1 + (j - 1) * self.sublen[i]) : int(
                                 self.len_[i] + m - 1 + j * self.sublen[i]
                             ),
                         ],
-                    )
+                    )[0]
                 )
         for i in range(1, m, 2):
             for j in range(1, self.nk + 1):
```

Comment:
Is this for-loop necessary?

Reply:
Yes,  I add comment on LSMOP's evaluate function for explanation. Beacuse all LSMOPs have same problem. Specifically， this function using uncertain variables (self.len_, self.sublen) to make sure start and length of array slice whcih making fori_loop disable.  Maybe "vmap" could replace these for-loops. I will try after in next merge.